syntax = "proto3";

//option java_multiple_files = true;
option java_package = "QXCommMngr";

package QXMSMsg;

enum MsgType {
    QX_MS_MSG_TYPE_UKNOWN                          	= 0;
	QX_MS_MSG_TYPE_SVR_HEALTH_REPORT				= 1;
	QX_MS_MSG_TYPE_STOP_WORKER						= 3;
	QX_MS_MSG_TYPE_STOP_WORKER_REPLY				= 4;
	QX_MS_MSG_TYPE_START_WORKER						= 5;
	QX_MS_MSG_TYPE_START_WORKER_REPLY				= 6;
}

message MsgPayload {
    uint32          transId                     = 1;
    uint64          timestamp                   = 2;
    string          bussinessKey                = 3;
    ServerInfo      serverInfo                  = 5;
    int32			errCode                     = 6;
    MsgBase         msgBase                     = 7;
}

message ServerInfo {
    uint32             serverId            = 1;
    string             serverName          = 2;
}

message MsgBase {
    uint32                 		msgType                 = 1;    // MsgType
	SvrHealthReport				svrHealthReport			= 2;	// QX_MS_MSG_TYPE_SVR_HEALTH_REPORT
}

message ClientInfo {
	uint32						clientId				= 1;
}

message SvrHealthReport {
	float						cpuUsage				= 1;
    float                       memUsage                = 2;
	repeated ClientInfo			clientInfo				= 3;
}
